File: backend/src/main/java/net/tape/api/VideoDtoV1.java
package net.tape.api;

/** Public (v1) DTO. */
public record VideoDtoV1(
    String slug,
    String title,
    String description,
    java.time.Instant publishedAt,
    Integer durationSec,
    String durationLabel,
    String showName,
    String categoryName,
    String views,
    long viewCount,
    java.util.List<String> tags,
    java.util.List<SecurityDtoV1> securities,
    java.util.List<PersonRefDtoV1> people,
    String imageUrl) {}

File: web/src/api/components/schemas/VideoDtoV1.ts
import { Option } from 'fp-ts/lib/Option';
import { string, Int, array, type } from 'io-ts';
import { DateFromISOString } from 'io-ts-types/lib/DateFromISOString';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type VideoDtoV1 = {
	slug: Option<string>;
	title: Option<string>;
	description: Option<string>;
	publishedAt: Option<Date>;
	durationSec: Option<Int>;
	durationLabel: Option<string>;
	showName: Option<string>;
	categoryName: Option<string>;
	tags: Option<Array<string>>;
	imageUrl: Option<string>;
};
export const VideoDtoV1IO = type(
	{
		slug: optionFromNullable(string),
		title: optionFromNullable(string),
		description: optionFromNullable(string),
		publishedAt: optionFromNullable(DateFromISOString),
		durationSec: optionFromNullable(Int),
		durationLabel: optionFromNullable(string),
		showName: optionFromNullable(string),
		categoryName: optionFromNullable(string),
		tags: optionFromNullable(array(string)),
		imageUrl: optionFromNullable(string),
	},
	'VideoDtoV1',
);
